spec/models/ont/plate_with_samples_factory_spec.rb

```diff
@@ -0,0 +1,357 @@
+          expect(factory.errors.full_messages.count).to eq(1)
+          expect(factory.errors.full_messages).to contain_exactly("Import was not successful: #{message}")
+        end
+
+        it 'with failed well insert' do
+          message = 'this is a well error'
+          allow(Well).to receive(:insert_all!).and_raise(message)
+          factory = Ont::PlateWithSamplesFactory.new
+          factory.process
+          expect(factory.save).to be_falsey
+          expect(factory.errors.full_messages.count).to eq(1)
+          expect(factory.errors.full_messages).to contain_exactly("Import was not successful: #{message}")
+        end
+
+        it 'with failed request insert' do
+          message = 'this is a request error'
+          allow(Ont::Request).to receive(:insert_all!).and_raise(message)
+          factory = Ont::PlateWithSamplesFactory.new
+          factory.process
+          expect(factory.save).to be_falsey
+          expect(factory.errors.full_messages.count).to eq(1)
+          expect(factory.errors.full_messages).to contain_exactly("Import was not successful: #{message}")
+        end
+
+        it 'with failed container_material insert' do
+          message = 'this is a container_material error'
+          allow(ContainerMaterial).to receive(:insert_all!).and_raise(message)
+          factory = Ont::PlateWithSamplesFactory.new
+          factory.process
+          expect(factory.save).to be_falsey
+          expect(factory.errors.full_messages.count).to eq(1)
+          expect(factory.errors.full_messages).to contain_exactly("Import was not successful: #{message}")
+        end
+
+        it 'with failed tag_taggable insert' do
+          message = 'this is a tag_taggable error'
+          allow(TagTaggable).to receive(:insert_all!).and_raise(message)
+          factory = Ont::PlateWithSamplesFactory.new
+          factory.process
+          expect(factory.save).to be_falsey
+          expect(factory.errors.full_messages.count).to eq(1)
+          expect(factory.errors.full_messages).to contain_exactly("Import was not successful: #{message}")
+        end
+      end
+
+      context 'failed transaction' do
+        before do
+          mock_valid_plate_factory
+          allow(Plate).to receive(:insert_all!).and_raise('this is an error')
+          factory = Ont::PlateWithSamplesFactory.new
+          factory.process
+          factory.save
+        end
+  
+        it 'does not insert any plates' do
+          expect(Plate.all.count).to eq(0)
+        end
+  
+        it 'does not insert any wells' do
+          expect(Well.all.count).to eq(0)
+        end
+  
+        it 'does not insert any container_materials' do
+          expect(ContainerMaterial.all.count).to eq(0)
+        end
+  
+        it 'does not insert any ont_requests' do
+          expect(Ont::Request.all.count).to eq(0)
+        end
+  
+        it 'does not insert any tag_taggables' do
+          expect(TagTaggable.all.count).to eq(0)
+        end
```

Comment:
It's good to see this being tested, though I'd like to see that all these database assertions are true, no matter which insert failed.  It might be that if you let the plate insert, and the wells, and the requests, but not the tag_taggables, that you have half the items in the database as this isn't asserted anywhere.